core/trino-main/src/main/java/io/trino/execution/QueryPreparer.java

```diff
@@ -65,6 +68,19 @@ public PreparedQuery prepareQuery(Session session, Statement wrappedStatement)
             prepareSql = Optional.of(session.getPreparedStatementFromExecute(executeStatement));
             statement = sqlParser.createStatement(prepareSql.get(), createParsingOptions(session));
         }
+        else if (statement instanceof ExecuteImmediate executeImmediateStatement) {
+            // HACK: pad the embedded statement with white space, so that it starts at the same position
+            //       as in the original EXECUTE IMMEDIATE statement, in order to get meaningful parse errors
+            int line = executeImmediateStatement.getStatementLocation().map(NodeLocation::getLineNumber).orElse(1);
+            int column = executeImmediateStatement.getStatementLocation().map(NodeLocation::getColumnNumber).orElse(1);
+            String prefix = "\n".repeat(line - 1) + " ".repeat(column); // account for the starting single quote
```

Comment:
Hmm... this is not great. We can deal with the parsing errors by catching the exception and rewriting the line/column.

I need to think about how to handle semantic or execution errors, though.

Reply:
Well, I'm not very proud of this, but I couldn't figure out a way to handle it when catching the exception.  It's not obvious how to compute the correct position when the embedded SQL statement spans more than one line.  Also, this hack seems to handle semantic exceptions, as well (see the test in my fixup commit).

Reply:
Are you fine with this approach for now?

Reply:
It’s probably ok for now, but you might want to check for downstream quirks such as how the query is being logged, for instance.

Also, add a TODO to fix it. A more robust approach would be to handle this all within the analyzer by having dedicated logic to deal with the ExecuteImmediate node there instead of doing it inside QueryPreparer.

Reply:
I have added a TODO.  I tried to add a test showing that the original query text is preserved in the `QueryCompletedEvent`, but I was not able to get event listeners invoked in tests (`QueryMonitor#queryCompletedEvent` does not get invoked by the test runners).  I was able to confirm through a manual test that the original query text is preserved.